src/mplfinance/_arg_validators.py

```diff
@@ -110,14 +110,34 @@ def _mav_validator(mav_value):
     '''
     if isinstance(mav_value,int) and mav_value > 1:
         return True
-    elif not isinstance(mav_value,tuple) and not isinstance(mav_value,list):
+    elif not isinstance(mav_value,(tuple,list,dict)):
         return False
 
-    if not len(mav_value) < 8:
-        return False
-    for num in mav_value:
-        if not isinstance(num,int) and num > 1:
+    if isinstance(mav_value,dict):
+        if 'period' not in mav_value or not isinstance(mav_value['period'],(tuple,list,dict)):
+            return False
+        if 'shift' in mav_value:
+            if not isinstance(mav_value['shift'],(tuple,list,dict)):
```

Comment:
ditto here: `(tuple,list)` ( *not* `(tuple,list,dict)`) )?

(or maybe it should be `(tuple,list,int)` ??   See also my general comments below).

Reply:
good catch.  I feel silly for that mistake.  Fixed